### ISSUE TITLE
error in using fbthrift

### ISSUE THREAD

**Issue by 1a1a11a** [microsoft/vcpkg#18249]
I have installed fbthrift, but when I use it I got the following error 

```
CMake Error at /PATH/vcpkg/installed/x64-linux/share/fbthrift/FBThriftConfig.cmake:24 (message):
  File or directory
  /PATH/vcpkg/installed/x64-linux/lib/cmake/fbthrift
  referenced by variable FBTHRIFT_CMAKE_DIR does not exist !
Call Stack (most recent call first):
  /PATH/vcpkg/installed/x64-linux/share/fbthrift/FBThriftConfig.cmake:47 (set_and_check)
  /PATH/vcpkg/scripts/buildsystems/vcpkg.cmake:772 (_find_package)
  CMakeLists.txt:98 (find_package)
``` 

I don't find a cmake file for fbthrift, can you help me? 

I have tried on macOSX, ubuntu18 and they all have the same problem.

**Comment by NancyLi1013** [comment]
Hi @1a1a11a 

Thanks for posting this issue. I was blocked to install `fbthrift `on all platforms. 

Can you please help provide the files listed in _share/fbthrift_ folder? Please also provide the usage of `fbthrift `you used in your project. Such as CMakeLists.txt.

Thanks.

**Comment by NancyLi1013** [comment]
After upgrading gcc version from 10.1.0 to 10.3.0, internal compiler error disappeared. See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=95579

```
code/vcpkg/buildtrees/fbthrift/src/d9021f5819-69b6281e02/thrift/lib/cpp2/transport/rocket/framing/Util.cpp:52:73: internal compiler internal compiler error: in sign_mask, at wide-int.h:855
error: in sign_mask, at wide-int.h:855
   52 |   const uint8_t frameType = frameTypeAndFlags >> Flags::frameTypeOffset();
      |                                                                         ^
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-10/README.Bugs> for instructions.
```
Now I can build `fbthrift ` successfully on Ubuntu 18.04 with gcc 10.3. 

Confirmed this is bug for fbthrift. I will look into this and try to fix it soon.

**Comment by 1a1a11a** [comment]
Hi @NancyLi1013, 
    Thank you for getting back quickly! I was using gcc 10.3.0 so installation is fine, but I cannot find it with cmake, a simplified example is shown below. And there is no `share/` directory under vcpkg, should I check other places? 

```
cmake_minimum_required (VERSION 3.10)
project (haha VERSION 0.1)

find_package(FBThrift REQUIRED) # must come after wangle

```

**Comment by NancyLi1013** [comment]
Thanks for your reply @1a1a11a.
 
It's unnecessary to check again. The *.cmake files are installed in _vcpkg_root_path/installed/share/${PORT}/_ folder. I have figured this out. So you don't need to provide the above information now.

I have submitted a PR #18264 to fix this issue. Can you please help test if this can work fine for you?

**Comment by 1a1a11a** [comment]
Thank you for the quick fix! 
On Ubuntu18.04 with gcc 10.3.0 and gcc 7.5.0, I am able to compile and find the library in cmake. 
However, I am not able to compile on MacOSX (I can compile previously). 

```
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/c++/4.2.1
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

The detailed error attached 
```
In file included from /Users/jason/software/vcpkg/buildtrees/fbthrift/src/269cfba3e3-f13d72fe1a.clean/thrift/lib/cpp2/server/RequestDebugLog.cpp:20:
/Users/jason/software/vcpkg/installed/x64-osx/include/folly/logging/xlog.h:1:1: error: unknown type name 'cloneCoalescedclion'
cloneCoalescedclion
^
In file included from /Users/jason/software/vcpkg/buildtrees/fbthrift/src/269cfba3e3-f13d72fe1a.clean/thrift/lib/cpp2/server/RequestDebugLog.cpp:20:
In file included from /Users/jason/software/vcpkg/installed/x64-osx/include/folly/logging/xlog.h:23:
In file included from /Users/jason/software/vcpkg/installed/x64-osx/include/folly/logging/LogStream.h:21:
In file included from /Users/jason/software/vcpkg/installed/x64-osx/include/folly/logging/LogCategory.h:21:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/list:201:1: error: expected unqualified-id
_LIBCPP_BEGIN_NAMESPACE_STD
^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__config:838:37: note: expanded from macro '_LIBCPP_BEGIN_NAMESPACE_STD'
#define _LIBCPP_BEGIN_NAMESPACE_STD namespace std { inline namespace _LIBCPP_ABI_NAMESPACE {
                                    ^
2 errors generated.
```

**Comment by NancyLi1013** [comment]
Thanks for your instant response. I was blocked by the installation of `folly` on macOS.

```
240 /Users/vcpkg/Nancy/vcpkg/buildtrees/folly/src/c305618e41-f9969d511a.clean/folly/synchronization/DistributedMutex-inl.h:1046:5: error: 'atom        ic_notify_one<unsigned long>' is unavailable
241 241     atomic_notify_one(state);
242 /Users/vcpkg/Nancy/vcpkg/buildtrees/folly/src/c305618e41-f9969d511a.clean/folly/synchronization/DistributedMutex-inl.h:1046:5: error: 'atom            ic_notify_one<unsigned long>' is unavailable
243 241 241     atomic_notify_one(state);
244 242 error: 'atomic_notify_one<unsigned long>' is unavailable
245 243     atomic_notify_one(state);
246 error: 'atomic_notify_one<unsigned long>' is unavailable
247     atomic_notify_one(state);
248     ^
249 /Users/vcpkg/Nancy/vcpkg/buildtrees/folly/src/c305618e41-f9969d511a.clean/folly/synchronization/DistributedMutex-inl.h:1562:5: note: in ins    tantiation of function template specialization 'folly::detail::distributed_mutex::wakeTimedWaiters<std::__1::atomic<unsigned long>>' reques    ted here
250     wakeTimedWaiters(&state_, proxy.timedWaiters_);
251     ^
252 /Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/atomic:2140:6: note: 'atomic_notify_one<unsigned long>' has been     explicitly marked unavailable here
253 void atomic_notify_one(atomic<_Tp>* __o) _NOEXCEPT
254      ^
255 1 error generated.
```

So I'm not sure the result on my machine. Do you use the same project with Linux? Is there something in .cpp?

**Comment by 1a1a11a** [comment]
(after some re-tests) I am able to install `folly` on macOSX (same config as above), after a few trials, I am able to install fbthrift now. 
I have taken a look of changed file, but with my limited knowledge of cmake, I don't see what could cause this. :( 
In conclusion, the patch works! Thank you! 

An irrelevant questions, the package from vcpkg is a little bit outdated and the repo I am compiling requires newer versions, what is the suggested way to try/install a newer version?

**Comment by NancyLi1013** [comment]
> An irrelevant questions, the package from vcpkg is a little bit outdated and the repo I am compiling requires newer versions, what is the suggested way to try/install a newer version?

There are two ways to install a newer version.

-  You can open a new issue to post the package that you want to update in vcpkg.
- If possible, you can submit a PR to update the package to the new version.

**Comment by autoantwort** [comment]
They will be updated in #17335

**Comment by NancyLi1013** [comment]
Thanks for your information @autoantwort.

**Comment by Cheney-W** [comment]
@1a1a11a 
The PR https://github.com/microsoft/vcpkg/pull/18264 has been close because the upstream has its own modification program. I want to confirm with you that whethre this issue still exists when don't use the change of the PR https://github.com/microsoft/vcpkg/pull/18264. Thank you!

**Comment by 1a1a11a** [comment]
I cannot test it now, so let's close it. Thank you!
